tools/editor/scene_tree_dock.cpp

```diff
@@ -1366,6 +1367,25 @@ void SceneTreeDock::_create() {
 
 		editor_data->get_undo_redo().create_action(TTR("Create Node"));
 
+		// Give the node a name with respect to the user's node name casing setting.
+		String new_name;
+		if (edited_scene) {
+			new_name = parent->validate_child_name(child);
+		} else {
+			new_name = child->get_type();
+		}
+		int name_casing = _get_node_name_casing("scenetree_editor/new_node_name_casing");
+		switch (name_casing) {
+
+			case NODE_NAME_CASING_CAMEL_CASE:
+				new_name[0] = towlower(new_name[0]); // First character should always be a letter here as it's a type name
```

Comment:
I think it would be better to use `String` methods for that instead of relying on `<wctype.h>`. I think this might work:
```
new_name[0] = new_name[0].to_lower();
```